parser.js

```diff
@@ -36,16 +36,24 @@ Parser.prototype._newPacket = function () {
 }
 
 Parser.prototype.parse = function (buf) {
+
   this._list.append(buf)
 
-  while ((this.packet.length != -1 || this._list.length > 0) &&
-         this[this._states[this._stateCounter]]()) {
-    this._stateCounter++
+  try {
+
+    while ((this.packet.length != -1 || this._list.length > 0) &&
+           (noError = this[this._states[this._stateCounter]]())) {
+      this._stateCounter++
 
-    if (this._stateCounter >= this._states.length) {
-      this._stateCounter = 0
+      if (this._stateCounter >= this._states.length) {
+        this._stateCounter = 0
+      }
     }
   }
+  catch (e) {
+
+    return this.emit('error', e);
+  }
```

Comment:
try-catch is slow, and this will really slow down everything.

Is this needed in the first place to fix this issue?